src/liballoc/collections/btree/node.rs

```diff
@@ -204,7 +207,7 @@ impl<K, V> Root<K, V> {
         Root {
             node: unsafe {
                 BoxedNode::from_ptr(NonNull::new_unchecked(
-                    &EMPTY_ROOT_NODE as *const _ as *const LeafNode<K, V> as *mut _,
+                    &EMPTY_ROOT_NODE as *const _ as *const NodeHeader<K, V> as *mut _,
```

Comment:
Nit: can now be simplified to `NonNull::from(&EMPTY_ROOT_NODE)`.

Reply:
The compiler and I think you're wrong there: something must be done to transmute the `NodeHeader<(), ()>` to `NodeHeader<K, V>`. Why we also have to tag it as mutable, beats me.

Earlier, I got off on a tangent to adjust `BoxedNode` to this actual type of `EMPTY_ROOT_NODE`, which implies removing K and V, or pretending to understand PhantomData, but I managed to squeeze that monster back into the box before it ate me.

Reply:
I forgot about the type parameters, mea culpa. `NonNull::from(&EMPTY_ROOT_NODE).cast()` should work, though.